types/amqp-connection-manager/index.d.ts

```diff
@@ -59,7 +59,7 @@ export interface AmqpConnectionManagerOptions {
  */
 export function connect(urls: string[], options?: AmqpConnectionManagerOptions): AmqpConnectionManager;
 
-export type SetupFunc = ((channel: ConfirmChannel, callback: (error?: Error) => void) => void) | ((channel: ConfirmChannel) => Promise<void>);
+export type SetupFunc = (channel: ConfirmChannel, callback?: (error?: Error) => void) => Promise<any> | void;
```

Comment:
I'm not sure I understand how this is an improvement.  It looks like it used to specify that you got a promise if and only if you didn't provide a callback, but now it can (ostensibly) do both (or neither)?  And why `Promise<any>`?  It looks like no value is expected.